### PR TITLE
add caicloud script & only sync ingress assigned to itself

### PR DESCRIPTION
**Pull request by mqliang**

### CODE REVIEW

File: controllers/nginx/caicloud-scripts/build-image.sh [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+echo "+++++ Start building nginx-ingress-controller"
+
+cd ${ROOT}
+
+# Build nginx-ingress-controller binary.
+CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o ./controllers/nginx/rootfs/nginx-ingress-controller ./controllers/nginx/pkg/cmd/controller
+
+# Build nginx-ingress-controller container.
+docker build -t caicloud/nginx-ingress-controller:${IMAGE_TAG} ./controllers/nginx/rootfs/.
+docker tag caicloud/nginx-ingress-controller:${IMAGE_TAG} index.caicloud.io/caicloud/nginx-ingress-controller:${IMAGE_TAG}
```
Comment (ScorpioCPH) [inline controllers/nginx/caicloud-scripts/build-image.sh]: nit: index.caicloud.io 这个已经停止维护了，可以删掉了

File: controllers/nginx/caicloud-scripts/build-image.sh [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+docker tag caicloud/nginx-ingress-controller:${IMAGE_TAG} cargo.caicloud.io/caicloud/nginx-ingress-controller:${IMAGE_TAG}
+docker tag caicloud/nginx-ingress-controller:${IMAGE_TAG} cargo.caicloudprivatetest.com/caicloud/nginx-ingress-controller:${IMAGE_TAG}
+
+cd - > /dev/null
+
+# Decide if we need to push images to caicloud registry.
+if [[ "$PUSH_TO_REGISTRY" == "Y" ]]; then
+  echo ""
+  echo "+++++ Start pushing nginx-ingress-controller"
+  docker push index.caicloud.io/caicloud/nginx-ingress-controller:${IMAGE_TAG}
```
Comment (ScorpioCPH) [inline controllers/nginx/caicloud-scripts/build-image.sh]: nit: rm index.caicloud.io

File: controllers/nginx/caicloud-scripts/build-image.sh [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  docker push cargo.caicloud.io/caicloud/nginx-ingress-controller:${IMAGE_TAG}
+  docker push cargo.caicloudprivatetest.com/caicloud/nginx-ingress-controller:${IMAGE_TAG}
+fi
+
+echo "Successfully built docker image caicloud/nginx-ingress-controller:${IMAGE_TAG}"
+echo "Successfully built docker image index.caicloud.io/caicloud/nginx-ingress-controller:${IMAGE_TAG}"
```
Comment (ScorpioCPH) [inline controllers/nginx/caicloud-scripts/build-image.sh:69]: nit: s/index.caicloud.io/cargo.caicloud.io

**Review by ScorpioCPH — APPROVED** [review]
LGTM.

**Comment by mqliang** [comment]
rebase 到了上游最新分支，因为之前fork的有个bug https://github.com/kubernetes/ingress/issues/411

**Comment by kdada** [comment]
Do not merge the pr. I'm writing a new pr to replace it.

**Pull request closed without merging**